cmd/sippy/automatejira.go

```diff
@@ -0,0 +1,157 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/openshift/sippy/pkg/api"
+	"github.com/openshift/sippy/pkg/api/componentreadiness"
+	crtype "github.com/openshift/sippy/pkg/apis/api/componentreport"
+	"github.com/openshift/sippy/pkg/apis/cache"
+	bqcachedclient "github.com/openshift/sippy/pkg/bigquery"
+	"github.com/openshift/sippy/pkg/componentreadiness/jiraautomator"
+	"github.com/openshift/sippy/pkg/flags"
+	"github.com/openshift/sippy/pkg/util/sets"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	prowflagutil "sigs.k8s.io/prow/pkg/flagutil"
+)
+
+type AutomateJiraFlags struct {
+	BigQueryFlags                                *flags.BigQueryFlags
+	GoogleCloudFlags                             *flags.GoogleCloudFlags
+	CacheFlags                                   *flags.CacheFlags
+	ComponentReadinessFlags                      *flags.ComponentReadinessFlags
+	JiraOptions                                  prowflagutil.JiraOptions
+	SippyURL                                     string
+	ComponentWhiteListStr                        string
+	ComponentWhiteList                           sets.String
+	VariantBasedComponentRegressionThresholdStrs []string
+	VariantBasedComponentRegressionThresholds    map[jiraautomator.Variant]int
+	JiraAccount                                  string
+}
+
+func NewAutomateJiraFlags() *AutomateJiraFlags {
+	return &AutomateJiraFlags{
+		BigQueryFlags:           flags.NewBigQueryFlags(),
+		GoogleCloudFlags:        flags.NewGoogleCloudFlags(),
+		CacheFlags:              flags.NewCacheFlags(),
+		ComponentReadinessFlags: flags.NewComponentReadinessFlags(),
+		VariantBasedComponentRegressionThresholds: map[jiraautomator.Variant]int{},
+	}
+}
+
+func (f *AutomateJiraFlags) BindFlags(fs *pflag.FlagSet) {
+	f.BigQueryFlags.BindFlags(fs)
+	f.GoogleCloudFlags.BindFlags(fs)
+	f.CacheFlags.BindFlags(fs)
+	f.ComponentReadinessFlags.BindFlags(fs)
+	f.JiraOptions.AddFlags(flag.CommandLine)
+	fs.AddGoFlagSet(flag.CommandLine)
+	fs.StringVar(&f.SippyURL, "sippy-url", f.SippyURL, "The Sippy URL prefix to be used to generate sharable Sippy links")
+	fs.StringVar(&f.ComponentWhiteListStr, "component-white-list", f.ComponentWhiteListStr, "The whitelist of comma seperated jira components to file issues against")
+	fs.StringArrayVar(&f.VariantBasedComponentRegressionThresholdStrs, "variant-based-component-regression-threshold", f.VariantBasedComponentRegressionThresholdStrs, "A list of variants which we will check for a column of red cells in component readiness report. If the number of red cells of a relevant column is over the threshold, we will a create jira card for the component corresponding to the specified variant (e.g. Bare Metal Hardware Provisioning for metal platform. The format of string is [variant]:[value]:[threshold] (e.g. Platform:metal:3)")
```

Comment:
I don't really like this name VariantBasedComponentRegressionThresholdStrs. Any better suggestions?

Reply:
How about ColumnThresholds

Reply:
The data in this command line option sounds like it's starting to get into config file territory.

Reply:
We should discuss about this too. Right now it is hardcoded in the code.